packages/liferay-npm-scripts/src/utils/getPaths.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+const expandGlobs = require('../utils/expandGlobs');
+const filterChangedFiles = require('../utils/filterChangedFiles');
+const filterGlobs = require('../utils/filterGlobs');
+const findRoot = require('../utils/findRoot');
+const log = require('../utils/log');
+const preprocessGlob = require('../utils/preprocessGlob');
+const readIgnoreFile = require('../utils/readIgnoreFile');
+
+/**
+ * Given a set of `globs`, the file `extensions` we are interested
+ * in, and the name of an `ignoreFile` (eg. '.prettierignore' or
+ * '.eslintignore'), returns a list of matching files on the file
+ * system.
+ */
+function getPaths(globs, extensions, ignoreFile) {
+	const root = findRoot();
+
+	ignoreFile = path.join(root || '.', ignoreFile);
+
+	const ignores = fs.existsSync(ignoreFile) ? readIgnoreFile(ignoreFile) : [];
+
+	// Turn "{src,test}/*" into ["src/*", "test/*"]:
+	globs = [].concat(...globs.map(glob => preprocessGlob(glob)));
```

Comment:
Once we're on a newer NodeJS version, this could use `Array.prototype.flatMap` instead.